packages/zipkin/src/internalRepresentations.js

```diff
@@ -29,6 +30,17 @@ Endpoint.prototype.toJSON = function toJSON() {
   };
 };
 
+Endpoint.createEndpoint = function createEndpoint({serviceName, host, port}, callback) {
+  getAddressByName(host, addr => {
```

Comment:
> ps it seems like this should just affect the host field, not the entire endpoint.

Could you describe this in more details, I'm not sure I've caught the point. 

We have to have a place where `host` will be set correctly. We can do this in `BachRecorder` and leave `internalRepresentations` completely free from async host-to-ip knowledge. 

Worth saying, I don't have any strong opinion about this part, so suggestions are totally welcome.